addons/stock/models/stock_rule.py

```diff
@@ -217,6 +217,8 @@ def _run_pull(self, procurements):
 
         # Prepare the move values, adapt the `procure_method` if needed.
         for procurement, rule in procurements:
+            if procurement.product_id.type not in ('consu', 'product'):
```

Comment:
I would directly add this check in run since we won't use a service in _run_buy or _run_manufacture.

Reply:
Although it might not be expected to work with purchase, you can buy a service (IIRC, that actually lead to the split purchase / purchase_stock). I'm not sure if this would work, but I don't think filtering the procurements too early is a good thing.

I'd rather fix directly where services clearly don't make sense, such as stock.

Reply:
in the "purchase mto" feature that led to the split purchase/purchase_stock, the procurements are bypassed

Reply:
The goal here was to avoid filtering the procurements too early, for future developments and third-party addons. If you think that procurements only make sense for consumable / stockable products, fine for me to filter them out at the `run` level.